data/csv.go

```diff
@@ -0,0 +1,32 @@
+package data
+
+import (
+	"encoding/csv"
+	"log"
+	"os"
+)
+
+type csvDataSource struct {
+	// Collection is a file path to a CSV file
+	Collection string
+}
+
+func NewCSVDataSource(csvPath string) csvDataSource {
+	return csvDataSource{Collection: csvPath}
+}
+
+func (ds csvDataSource) ReadCollection() [][]string {
+	file, err := os.Open(ds.Collection)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer file.Close()
+
+	csvReader := csv.NewReader(file)
+	records, err := csvReader.ReadAll()
+	if err != nil {
+		log.Fatalln(err)
```

Comment:
Be careful with using `log.FatalX` (check the previous comment). Also, when using defer (line 23), if line 28 executes, the defer line will not run and could create some problems with leaving the file open.